src/libs/WorkspacesSettingsUtils.ts

```diff
@@ -175,12 +180,18 @@ function getWorkspacesBrickRoads(): Record<string, BrickRoad> {
         }
     });
 
-    Object.values(allReports).forEach((report) => {
+    const actionsByReport: OnyxCollection<ReportActions> = {};
+    Object.keys(reportActions ?? {}).forEach((key) => {
+        const reportID = CollectionUtils.extractCollectionItemID(key as typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS);
+        actionsByReport[reportID] = reportActions?.[key] ?? {};
```

Comment:
@eh2077 I remember I added that if (!reportActions) early return because otherwise `reportActions.[key]` was throwing possibly `undefined` type error.

I removed the check and instead added optional chaining and fallback to empty object like so: `reportActions?.[key] ?? {}`. Let me know if it's good like this!

Reply:
@ikevin127 I'm confused here. Let's say `reportActions` is undefined, it shouldn't enter the foreach loop right?

Reply:
> Let's say reportActions is undefined, it shouldn't enter the foreach loop right?

@eh2077 Nope, since we fallback to empty object `Object.keys(reportActions ?? {}).forEach...` it enters the foreach loop.

Then: `actionsByReport[reportID]` is expecting type `Record<string, ReportActions | null>` because it's defined above as:
```ts
const actionsByReport: OnyxCollection<ReportActions> = {};
```

If I remove optional chaining from `reportActions?.[key]` I get the following type error:

<img width="509" alt="Screenshot 2024-04-15 at 18 59 44" src="https://github.com/Expensify/App/assets/56457735/0e33905b-3e10-4803-ba5e-ffcc7128182a">

If I keep optional chaining but don't have the empty object fallback I get the following type error from `actionsByReport[reportID]`:

<img width="703" alt="Screenshot 2024-04-15 at 19 00 31" src="https://github.com/Expensify/App/assets/56457735/b541ae01-9610-4500-91c5-e39c1ce7a669">

So we can either keep the if (!reportActions) early return or the optional chaining + fallback to empty object.
If you see other ways to do this please let me know!

Reply:
Oh, I see! I like `reportActions?.[key]`. Thank you for you patience.